modules/apps/foundation/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/buffer/BufferedIndexerInvocationHandler.java

```diff
@@ -201,10 +201,17 @@ protected void bufferRequest(
 
 		ClassedModel classedModel = (ClassedModel)baseModel.clone();
 
-		IndexerRequest indexerRequest = new IndexerRequest(
-			methodKey.getMethod(), classedModel, _indexer);
+		Long classPK = (Long)classedModel.getPrimaryKeyObj();
 
-		doBufferRequest(indexerRequest, indexerRequestBuffer);
+		if (object instanceof ResourcedModel) {
+			ResourcedModel resourcedModel = (ResourcedModel)object;
+
+			classPK = resourcedModel.getResourcePrimKey();
+		}
+
+		bufferRequest(
```

Comment:
Replacing object with classPK / resourcePrimKey cannot be done in delete operations, because there is no index.delete(className,classPK) method, so you have to check method name is "reindex" and only do it in that case.

See this: https://github.com/giros/liferay-portal/blob/540129e26a0d9276e95c49569807cf5913b36621/modules/apps/foundation/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/buffer/BufferedIndexerInvocationHandler.java#L105-L118

I think you can remove the `args[0] instanceof ResourcedModel` there and add your logic that retrieves getPrimaryKeyObj or getResourcePrimKey (ResourcedModel) depending on object type

Reply:
Thanks for the heads up @jorgediaz-lr, I'll revisit my code according to this.